recipes/default.rb

```diff
@@ -21,7 +21,7 @@
 
 # Start+Enable the lvmetad service on RHEL7, it is required by default
 # but not automatically started
-if node['platform_family'] == 'rhel' && node['platform_version'].to_i >= 7
+if node['platform_family'] == 'rhel' && node[platform] != 'amazon' && node['platform_version'].to_i >= 7
```

Comment:
Additional investigation.

The reason this didn't fail for `amazon linux 2016` is because `amazon linux 2016` doesn't have the `use_lvmetad` config set to 1, hence it skipped the step to enable/start the `lvm2-lvmetad` service
```bash
[ec2-user@ip-172-31-28-89 ~]$ cat /etc/os-release
NAME="Amazon Linux AMI"
VERSION="2016.09"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2016.09"
PRETTY_NAME="Amazon Linux AMI 2016.09"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2016.09:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
[ec2-user@ip-172-31-28-89 ~]$ sudo /sbin/lvm dumpconfig global/use_lvmetad
use_lvmetad=0
```

However for amazon linux 2017 if has the `use_lvmetad` config set to 1. 
```bash
[ec2-user@ip-172-31-20-128 ~]$ cat /etc/os-release
NAME="Amazon Linux AMI"
VERSION="2017.03"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2017.03"
PRETTY_NAME="Amazon Linux AMI 2017.03"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2017.03:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
[ec2-user@ip-172-31-20-128 ~]$ sudo /sbin/lvm dumpconfig global/use_lvmetad
use_lvmetad=1
```
hence it will try to start the `lvm2-lvmetad` service (as it requires it to have lvm working). But it fails because amazon linux 2017 chooses to use `init.d` instead of `systemd`